topostats/processing.py

```diff
@@ -200,74 +197,71 @@ def run_grains(  # noqa: C901
-                            direction_grain_crops: GrainCropsDirection = grains.image_grain_crops.above
                         else:
-                            direction_grain_crops: GrainCropsDirection = grains.image_grain_crops.below
-                        if direction_grain_crops is not None:
-                            LOGGER.info(f"[{filename}] : Plotting individual grain masks")
-                            for grain_number, grain_crop in direction_grain_crops.crops.items():
-                                # Crop image plot
-                                crop_image = grain_crop.image
-                                plotting_config["plot_dict"]["grain_image"][
-                                    "filename"
-                                ] = f"{filename}_grain_{grain_number}"
-                                plotting_config["plot_dict"]["grain_image"]["output_dir"] = grain_out_path_direction
-                                Images(
-                                    data=crop_image,
-                                    **plotting_config["plot_dict"]["grain_image"],
-                                ).plot_and_save()
-                                # Grain mask plot
-                                crop_mask = grain_crop.mask
-                                plotting_config["plot_dict"]["grain_mask"]["output_dir"] = grain_out_path_direction
-                                if len(crop_mask.shape) == 3:
-                                    # Tensor, iterate over channels
-                                    for tensor_class in range(1, crop_mask.shape[2]):
-                                        plotting_config["plot_dict"]["grain_mask"][
-                                            "filename"
-                                        ] = f"{filename}_grain_mask_{grain_number}_class_{tensor_class}"
-                                        Images(
-                                            data=crop_image,
-                                            masked_array=crop_mask[:, :, tensor_class],
-                                            **plotting_config["plot_dict"]["grain_mask"],
-                                        ).plot_and_save()
-                                else:
-                                    # 2D array
+                            # 2D array
+                            LOGGER.debug(f"[{filename}] : Plotting {plot_name} image")
+                            plotting_config["plot_dict"][plot_name]["output_dir"] = grain_out_path_direction
+                            Images(
+                                data=np.zeros_like(array),
+                                masked_array=array,
+                                **plotting_config["plot_dict"][plot_name],
+                            ).plot_and_save()
+                    # Plot individual grain masks
+                    if direction == "above":
+                        direction_grain_crops: GrainCropsDirection = grains.image_grain_crops.above
```

Comment:
Personally I find it easier and more logically consistent when instantiated objects follow the same order as the class they represent so pedantically suggest...

```suggestion
                        grain_crops_direction: GrainCropsDirection = grains.image_grain_crops.above
```


...which obviously needs cascading through (the language server in your IDE should make such a change trivial and is more reliable than a find/replace).